islands/VoteBtn.tsx

```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import type { Item, Vote } from "@/utils/db.ts";
+
+export default function VoteBtn(
+  props: { item: Item; votes: Vote[] | [] },
+) {
+  const upvoted = !!props.votes.filter((item) => item.itemId === props.item.id)
+    .length;
+  const voteId = props.votes.find((item) => item.itemId === props.item.id)?.id;
+  return (
+    <button
+      class={`cursor-pointer mr-2 ${
+        upvoted ? "text-pink-700" : "text-gray-300"
+      }`}
+      type="submit"
+      onClick={() => {
+        const url = new URL("/api/vote", location.href);
+        url.searchParams.set("to", props.item.id);
+        let method = "POST";
+        if (upvoted) {
+          url.searchParams.set("vote", voteId as string);
+          method = "DELETE";
+        }
+        fetch(url, {
+          method,
+          credentials: "same-origin",
+        }).then((res) => {
+          if (res.ok) {
+            location.assign(window.location.href);
+          } else {
+            location.assign("/login");
+          }
+        });
```

Comment:
We don't need this entire `.then()` method as it is now. The endpoint itself will take care of any unauthenticated requests itself. Instead, the response received from `fetch()` should be used to set the value of the `upvoted` variable.